### PR TITLE
feat: module tools doc migrate

### PR DESCRIPTION
**Pull request by sanyuan0704**
# PR Details

模块工程文档迁移，主要改动:

- 去掉 .island 目录 
- 引入 @modern-js/doc-tools，并添加 modern.config.ts 配置文件
- 添加 theme 目录引入自定义 css
- 图片上传 CDN 处理
- 自动 sidebar 生成

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: website/module-tools/modern.config.ts [hunk 1]
```diff
@@ -268,30 +191,77 @@ function getNavbar(lang: 'zh' | 'en') {
     {
       text: getText('指南', 'Guide'),
       link: getLink('/guide/intro/welcome'),
-      activeMatch: '/guide/'
+      activeMatch: '^/guide/',
```
Comment (10Derozan) [inline website/module-tools/modern.config.ts:194]: 这个^的作用是啥？

File: website/module-tools/modern.config.ts [hunk 2]
```diff
@@ -106,68 +23,74 @@ function getSidebar(lang: 'zh' | 'en') {
...
-        ]
+        ],
       },
       {
         text: getText('基础使用', 'Basic Guide'),
         items: [
           {
             text: getText('开始之前', 'Before getting started'),
-            link: getLink('/guide/basic/before-getting-started')
+            link: getLink('/guide/basic/before-getting-started'),
```
Comment (10Derozan) [inline website/module-tools/modern.config.ts:46]: 这个doc-tools应该已经支持自动生成sidebar了吧，迁移主站文档应该有遇到，之前是手动写的，可以改一下，目前也是按照目录划分的

File: website/module-tools/docs/zh/guide/basic/test-your-project.mdx [hunk 3]
```diff
@@ -6,20 +6,19 @@
 
 想要使用项目的测试功能，需要确保项目中包含以下依赖：
 
-* `"@modern-js/plugin-testing"`
+- `"@modern-js/plugin-testing"`
```
[the remark this answers is not in the document]
Comment (sanyuan0704) [reply website/module-tools/docs/zh/guide/basic/test-your-project.mdx:9]: 不是，我本地有 markdown lint 工具

File: website/module-tools/modern.config.ts [hunk 4]
```diff
@@ -1,101 +1,18 @@
...
   const getText = (cnText: string, enText: string) => (cn ? cnText : enText);
   return { getText, getLink };
 }
 
-const isDevCommand = process.argv.includes('dev');
-
-export default defineConfig({
-  lang: 'en-US',
-  title: 'Module tools',
-  base: isDevCommand ? '' : '/module-tools/',
```
[the remark this answers is not in the document]
Comment (sanyuan0704) [reply website/module-tools/modern.config.ts:22]: 是的， 我稍等补充上 base 相关的配置

File: website/module-tools/modern.config.ts (hunk 1 above)
Comment (sanyuan0704) [reply website/module-tools/modern.config.ts:194]: 正则，强制匹配以 /guide/ 开头。我发现有些路由 guide 和 plugins 都包含，导航栏会高亮两个tab

File: website/module-tools/modern.config.ts (hunk 2 above)
Comment (sanyuan0704) [reply website/module-tools/modern.config.ts:46]: 也可以，我看看怎么接入

File: website/module-tools/modern.config.ts [hunk 5]
```diff
@@ -1,101 +1,18 @@
...
-        lastUpdatedText: 'Last Updated',
-        editLink: {
-          pattern:
-            'https://github.com/modern-js-dev/modern.js/tree/main/website/module-tools/docs/:path',
-          text: '📝 Edit this page on GitHub'
-        }
-      }
-    },
-    outlineTitle: 'ON THIS PAGE',
-    socialLinks: [
```
[the remark this answers is not in the document]
Comment (sanyuan0704) [reply website/module-tools/modern.config.ts:84]: 对的，默认值就没必要配了

**Review by 10Derozan — APPROVED** [review]

**Pull request merged**